File: set-mismatch.rs
impl Solution {
  pub fn find_error_nums(nums: Vec<i32>) -> Vec<i32> {
    let mut ans = vec![0; 2];

    let n = nums.len();
    let mut t = vec![0; n + 1];
    for i in 0..n {
      t[nums[i] as usize] += 1;
    }
    for i in 1..=n {
      if t[i] == 2 { ans[0] = i as i32; }
      if t[i] == 0 { ans[1] = i as i32; }
    }
    ans
  }
}

File: s/set-mismatch.cc
class Solution {
public:
  vector<int> findErrorNums(vector<int>& nums) {
    int N = nums.size();
    vector<int> T(N + 1);
    for (int e : nums) T[e]++;

    vector<int> ans(2);
    for (int i = 1; i <= N; ++i) {
      if (T[i] == 2) ans[0] = i;
      if (T[i] == 0) ans[1] = i;
    }
    return ans;
  }
};
